amsclient/utils_test.go

```diff
@@ -54,6 +58,103 @@ func MakeTokenString(typ string, life time.Duration) string {
 	return token.Raw
 }
 
+func equalClusterInfos(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+func TestGenerateSearchParameter(t *testing.T) {
+	orgID := "org123"
+	allowedStatuses := []string{"status1", "status2"}
+	disallowedStatuses := []string{"status3", "status4"}
+
+	expectedSearchQuery := "organization_id is 'org123' and cluster_id != '' and status in ('status1','status2') and status not in ('status3','status4')"
+
+	result := amsclient.GenerateSearchParameter(orgID, allowedStatuses, disallowedStatuses)
+
+	if result != expectedSearchQuery {
+		t.Errorf("Expected: %s, Got: %s", expectedSearchQuery, result)
+	}
+}
+
+func TestGenerateMulticLusterSearchQuery(t *testing.T) {
+	orgID := "org456"
+	clusterIDs := []string{"cluster4", "cluster5", "cluster6"}
+	allowedStatuses := []string{"status5", "status6"}
+	disallowedStatuses := []string{"status7", "status8"}
+
+	expectedSearchQuery := "organization_id is 'org456' and cluster_id in ('cluster4','cluster5','cluster6') and status in ('status5','status6') and status not in ('status7','status8')"
+
+	result := amsclient.GenerateMulticLusterSearchQuery(orgID, clusterIDs, allowedStatuses, disallowedStatuses)
+
+	if result != expectedSearchQuery {
+		t.Errorf("Expected: %s, Got: %s", expectedSearchQuery, result)
+	}
+}
+
+func TestFilterManagedClusters(t *testing.T) {
+	// Scenario 1: Empty input slice
```

Comment:
why not putting each scenario in a test case using `t.Run`?

Reply:
can do =)

Reply:
done